great_expectations/rule_based_profiler/rule_based_profiler.py

```diff
@@ -423,152 +480,187 @@ def _reconcile_rule(
+        builder configuration contributes to the parameter builders list of the rule if the corresponding parameter
+        builder name does not exist in the list of instantiated parameter builders of the rule; otherwise, once
+        instnatiated, it replaces the configuration associated with the original parameter builder having the same name.
+
+        :param rule: Profiler "rule", subject to parameter builder overrides
+        :param parameter_builder_configs: parameter builder configuration overrides, supplied in dictionary (configuration) form
+        :return: reconciled parameter builder configuration, returned in dictionary (configuration) form
+        """
         parameter_builder_config: dict
         for parameter_builder_config in parameter_builder_configs:
-            parameter_builder_name: str = parameter_builder_config["name"]
-            if parameter_builder_name in current_parameter_builders:
-                parameter_builder: ParameterBuilder = current_parameter_builders[
-                    parameter_builder_name
-                ]
-                effective_parameter_builder_configs.append(
-                    RuleBasedProfiler._reconcile_parameter_builder_config(
-                        parameter_builder=parameter_builder,
-                        parameter_builder_config=parameter_builder_config,
-                    )
-                )
-            else:
-                effective_parameter_builder_configs.append(parameter_builder_config)
+            validate_builder_config(builder_config=parameter_builder_config)
 
-        return effective_parameter_builder_configs
+        effective_parameter_builder_configs: Dict[str, dict] = {}
 
-    @staticmethod
-    def _reconcile_parameter_builder_config(
-        parameter_builder: ParameterBuilder,
-        parameter_builder_config: dict,
-    ) -> dict:
-        effective_parameter_builder_config: dict = {}
-        batch_request: Optional[
-            Union[BatchRequest, RuntimeBatchRequest, dict]
-        ] = parameter_builder_config.pop("batch_request", None)
-        if batch_request is None:
-            batch_request = get_batch_request_as_dict(
-                batch_request=parameter_builder.batch_request
+        current_parameter_builders: Dict[
+            str, ParameterBuilder
+        ] = rule._get_parameter_builders_as_dict()
+
+        parameter_builder_name: str
+        parameter_builder: ParameterBuilder
+        parameter_builder_as_dict: dict
+        for (
+            parameter_builder_name,
+            parameter_builder,
+        ) in current_parameter_builders.items():
+            parameter_builder_as_dict = parameter_builder.to_dict()
+            parameter_builder_as_dict[
+                "class_name"
+            ] = parameter_builder.__class__.__name__
+            parameter_builder_as_dict[
+                "module_name"
+            ] = parameter_builder.__class__.__module__
```

Comment:
Let's move this functionality to methods on the parameter builder class

Reply:
@cdkini Let us add this one to our discussion as well. Thanks!